vmpooler_client/command_parser.py

```diff
@@ -65,28 +65,67 @@ def __init__(self, argv):
     self._commands = {}
     self._sub_commands = {}
 
-  def add_command(self, cmd_name, desc):
+  def add_command(self, cmd_name, desc, func=None):
     """Create a top-level command.
 
     Args:
       cmd_name |str| = The name of the command.
       desc |str| = A description of the command.
+      func |func| = A function to use for commands that do not allow sub-commands.
+        If any value is supplied it is assumed that the command will never use sub-commands.
 
     Returns:
       None
 
     Raises:
-      |KeyError| Command name has already been defined.
+      |KeyError| = Command name has already been defined.
     """
 
     if cmd_name not in self._commands:
       command = self._sub_parsers.add_parser(cmd_name, description=desc)
 
       # Add a subparser for sub-commands under a top-level command. (i.e. "list" for "vm")
-      self._commands[cmd_name] = command.add_subparsers()
+      if func is None:
+        self._commands[cmd_name] = command.add_subparsers()
+
+        # Tack on this bit of meta-data to indicate that sub-commands are ALLOWED
+        self._commands[cmd_name]._allow_sub_cmd = True
+      else:
+        self._commands[cmd_name] = command
+        self._commands[cmd_name].set_defaults(func=func)
+
+        # Tack on this bit of meta-data to indicate that sub-commands are NOT allowed.
+        self._commands[cmd_name]._allow_sub_cmd = False
```

Comment:
I have a different strategy for doing this as well. I'll make the change and we can decide if it makes more sense.